File: elixir/lib/day1.ex
defmodule Day1a do
  def solve(input_path) do
    {tail, head} =
      input_path
      |> File.stream!()
      |> Enum.reduce({[], 0}, &reducer/2)

    all = [head | tail]

    answer_part1 = Enum.max(all)

    answer_part2 =
      all
      |> Enum.sort(:desc)
      |> Enum.take(3)
      |> Enum.sum()

    IO.puts("Answer to part 1: #{answer_part1}")
    IO.puts("Answer to part 2: #{answer_part2}")
  end

  defp reducer(line, {all_elves_totals, current_elf_calories}) when line == "\n",
    do: {[current_elf_calories | all_elves_totals], 0}

  defp reducer(line, {all_elves_totals, current_elf_calories}) do
    {calories, _} = Integer.parse(line)
    {all_elves_totals, current_elf_calories + calories}
  end
end

File: python/day1.py
def solve(path):
    f = open(path)
    lines = f.readlines()

    elf_calories = []
    total = 0
    for line in lines:
        if line == "\n":
            elf_calories.append(total)
            total = 0
        else:
            total += int(line)

    answer_part_1 = max(elf_calories)
    answer_part_2 = sum(sorted(elf_calories, reverse=True)[0:3])

    print(f"Part 1 answer: {answer_part_1}")
    print(f"Part 2 answer: {answer_part_2}")
